src/frontend/hooks/useSetHeader.tsx

```diff
@@ -39,9 +38,31 @@ export const useSetHeader = (
     backgroundColor,
     headerTintColor,
     headerShown,
-    headerLeft,
   } = titleOrOptions;
 
+  // For some reason, when headerTintColor is set, the back button is not getting the tint color. So I am setting it directly in custom header left. Header left should only be rewritten if there is a color. Otherwise, setting it as undefined, causes it to be removed.
+  if (!!headerTintColor) {
+    return React.useLayoutEffect(() => {
+      navigation.setOptions({
+        headerTitle: isMessageDescriptor(headerTitle)
+          ? t(headerTitle)
+          : headerTitle,
+        headerRight: headerRight,
+        headerStyle: {
+          backgroundColor: backgroundColor,
+        },
+        headerLeft: props => (
+          <CustomHeaderLeft
+            headerBackButtonProps={props}
+            tintColor={headerTintColor}
+          />
+        ),
+        headerTintColor: headerTintColor,
+        headerShown,
+      });
+    }, [navigation, t, titleOrOptions]);
```

Comment:
You [can't have a hook inside a conditional](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level), it will break.